tests/regression/test_slepc.py

```diff
@@ -0,0 +1,80 @@
+import pytest
+import numpy as np
+from firedrake import *
+from firedrake.petsc import PETSc
+
+
+def topetsc(A):
+    return A.M.handle
+
+
+def test_import():
+    from slepc4py import SLEPc
+
+
+def test_laplace_physical_ev(parallel=False):
+    from slepc4py import SLEPc
+
+    mesh = UnitSquareMesh(64, 64)
+    V = FunctionSpace(mesh, 'CG', 1)
+
+    u = TrialFunction(V)
+    v = TestFunction(V)
+
+    bc = DirichletBC(V, Constant(0.0), (1, 2, 3, 4))
+
+    # We just need the Stiffness and Mass matrix
+    a = inner(grad(u), grad(v))*dx
+    m = inner(u, v)*dx
+
+    A = topetsc(assemble(a, bcs=[bc]))
+    M = topetsc(assemble(m, bcs=[bc]))
+
+    # This shifts the "1.0" Eigenvalues out of the spectrum
+    # of interest (which is around 0.0 in this case).
+    vals = np.repeat(1E8, len(bc.nodes))
+    A.setValuesLocalRCV(bc.nodes.reshape(-1, 1),
+                        bc.nodes.reshape(-1, 1),
+                        vals.reshape(-1, 1))
+    A.assemble()
+
+    E = SLEPc.EPS()
+    E.create()
+    E.setOperators(A, M)
+    st = E.getST()
+    st.setType('sinvert')
+    kspE = st.getKSP()
+    kspE.setType('fgmres')
+    pc = kspE.getPC()
+    pc.setType('lu')
+    if parallel:
+        pc.setFactorSolverPackage('mumps')
```

Comment:
On travis we don't install petsc with support for mumps (since it takes rather a long time to build).  Is it possible to recast this in terms of an iterative solver (you're only actually looking at a few of the eigenmodes after all).